test/tla/ERC20.tla

```diff
@@ -0,0 +1,314 @@
+----------------------------- MODULE ERC20 ------------------------------------
+(*
+ * Modeling ERC20 tokens of Ethereum and the Approve-TransferFrom Attack:
+ *
+ * EIP-20: https://eips.ethereum.org/EIPS/eip-20
+ *
+ * Attack scenario:
+ * https://docs.google.com/document/d/1YLPtQxZu1UAvO9cZ1O2RPXBbT0mooh4DYKjA_jp-RLM/edit#
+ *
+ * This TLA+ specification is designed for model checking with Apalache.
+ * We do not model 256-bit integers here, as we are not interested in overflows.
+ *
+ * Igor Konnov, Gabriela Moreira, Informal Systems, 2021-2022
+ *)
+
+
+EXTENDS Integers, Apalache, Variants, ERC20_typedefs
+
+CONSTANTS
+    \* Set of all addresses.
+    \*
+    \* @type: Set(ADDR);
+    ADDR,
+    \* Set of the amounts to use.
+    \*
+    \* @type: Set(Int);
+    AMOUNTS
+
+VARIABLES
+    \* Token balance for every account. This is exactly as `balanceOf` in ERC20.
+    \*
+    \* @type: ADDR -> Int;
+    balanceOf,
+    \* Allowance to transfer tokens
+    \* from owner (1st element) by spender (2nd element).
+    \* This is exactly as `allowance` of ERC20.
+    \*
+    \* @type: <<ADDR, ADDR>> -> Int;
+    allowance
+
+\* Variables that model Ethereum transactions, not the ERC20 state machine.
+VARIABLES
+    \* Pending transactions to be executed against the contract.
+    \* Note that we have a set of transactions instead of a sequence,
+    \* as the order of transactions on Ethereum is not predefined.
+    \* To make it possible to submit two 'equal' transactions,
+    \* we introduce a unique transaction id.
+    \*
+    \* @type: Set(TX);
+    pendingTransactions,
+    \* The last executed transaction.
+    \*
+    \* @type: TX;
+    lastTx,
+    \* A serial number to assign unique ids to transactions
+    \* @type: Int;
+    nextTxId
+
+\* TX constructors
+
+\* @type: TX;
+None ==
+    Variant("None", [ id |-> 0, fail |-> FALSE ])
+
+\* @type: $transfer => TX;
+Transfer(fields) ==
+    Variant("Transfer", fields)
+
+\* @type: $approve => TX;
+Approve(fields) ==
+    Variant("Approve", fields)
+
+\* @type: $transferFrom => TX;
+TransferFrom(fields) ==
+    Variant("TransferFrom", fields)
+
+\* Initialize an ERC20 token.
+Init ==
+    /\ balanceOf = [a \in ADDR |-> 100]
```

Comment:
Shall we introduce a constant for the initial balance?

Reply:
I don't think so. As we were discussing yesterday, `Init` is also a form of parametrization, so that would be just a transfer of parameters IMO.

Reply:
I mean, `ERC20.tla` is parameterized with `ADDR` and `AMOUNTS`. This parameters are instantiated in `MC_ERC20.tla`. In this sense, we already have parameters in this spec, so why not introducing another one?

Reply:
Ok, I think it makes sense to prioritize parametrization by constants when possible (as oppose to defining multiple `Init`s). I'll add the parameter.

Reply:
I've added a constant of type `ADDR -> Int` to hold the initial balances for all accounts, so we can define accounts with different initial balances.